tests/phpunit/tests/post/query.php

```diff
@@ -758,4 +758,612 @@ public function test_found_posts_should_be_integer_even_if_found_posts_filter_re
+	 * @dataProvider dataFields
+	 *
+	 * @param string $fields
+	 */
+	public function test_found_posts_are_correct_for_meta_queries( $fields ) {
+		self::factory()->post->create_many( 5 );
+		$ids = self::factory()->post->create_many( 5 );
+
+		foreach ( $ids as $id ) {
+			add_post_meta( $id, 'my_meta', 'foo' );
+		}
+
+		$q = new WP_Query(
+			array(
+				'fields'         => $fields,
+				'posts_per_page' => 2,
+				'meta_key'       => 'my_meta',
+			)
+		);
+
+		$this->assertSame( 2, $q->post_count, self::get_count_message( $q ) );
+		$this->assertSame( 5, $q->found_posts, self::get_count_message( $q ) );
+		$this->assertEquals( 3, $q->max_num_pages, self::get_count_message( $q ) );
+	}
+
+	/**
+	 * @ticket 47280
+	 * @group ms-required
+	 * @dataProvider dataFields
+	 *
+	 * @param string $fields
+	 */
+	public function test_found_posts_are_correct_when_switching_between_sites( $fields ) {
+		$blog = self::factory()->blog->create();
+		self::factory()->post->create_many( 5 );
+
+		$q = new WP_Query(
+			array(
+				'fields'         => $fields,
+				'posts_per_page' => 2,
+			)
+		);
+
+		// Switch to another site.
+		switch_to_blog( $blog );
+
+		// Count the posts from the original site. This works because the SQL query
+		// and its table names has already been formed during the original query.
+		$post_count    = $q->post_count;
+		$found_posts   = $q->found_posts;
+		$max_num_pages = $q->max_num_pages;
+
+		// Switch back.
+		restore_current_blog();
+
+		$this->assertSame( 2, $post_count, self::get_count_message( $q ) );
+		$this->assertSame( 5, $found_posts, self::get_count_message( $q ) );
+		$this->assertEquals( 3, $max_num_pages, self::get_count_message( $q ) );
+	}
+
+	/**
+	 * @ticket 47280
+	 * @expectedDeprecated The posts_request filter
+	 */
+	public function test_posts_are_counted_with_select_found_rows_when_query_includes_sql_calc_found_rows() {
+		// Create five published posts.
+		self::factory()->post->create_many( 5 );
+		// Create ten draft posts.
+		self::factory()->post->create_many(
+			10,
+			array(
+				'post_status' => 'draft',
+			)
+		);
+
+		add_filter(
+			'posts_request',
+			function( $request ) {
+				global $wpdb;
+
+				return "
+					SELECT SQL_CALC_FOUND_ROWS {$wpdb->posts}.ID
+					FROM {$wpdb->posts}
+					WHERE 1=1
+					AND {$wpdb->posts}.post_type = 'post'
+					AND {$wpdb->posts}.post_status = 'draft'
+					ORDER BY {$wpdb->posts}.post_date
+					DESC LIMIT 0, 2
+				";
+			}
+		);
+
+		$q = new WP_Query(
+			array(
+				'posts_per_page' => 2,
+			)
+		);
+
+		// These results should now reflect the results for draft posts, as set by the filter
+		$this->assertStringContainsString( 'SELECT FOUND_ROWS()', $q->count_request );
+		$this->assertSame( 2, $q->post_count, self::get_count_message( $q ) );
+		$this->assertSame( 10, $q->found_posts, self::get_count_message( $q ) );
+		$this->assertEquals( 5, $q->max_num_pages, self::get_count_message( $q ) );
+	}
+
+	public function dataFields() {
```

Comment:
```suggestion
	public function data_fields() {
```
Per [#53119](https://core.trac.wordpress.org/ticket/53119), I'd recommend using the `data_` prefix along with `snake_case` as this has become common practice within the test suite.